extend/flytekit-xgboost/flytekitplugins/xgboost/task.py

```diff
@@ -159,49 +191,36 @@ def test(self, booster_model: str, dtest: xgboost.DMatrix, **kwargs) -> List[flo
         return y_pred
 
     def execute(self, **kwargs) -> Any:
-        if not all(x in kwargs for x in ["train", "test"]):
-            raise ValueError(
-                "Must have train and test inputs; 'train' and 'test' should be the actual parameter keys"
-            )
-
-        # replace model_parameters, hyper_parameters, label_column with the user-given values
-        for key, type in get_type_hints(XGBoostParameters).items():
-            if key in kwargs:
-                if issubclass(self.python_interface.inputs[key], type):
-                    # calling `Updateable.update()` to update the hyperparameters
-                    self._config.update({key: kwargs[key]})
-                else:
-                    raise TypeError(f"{key} has to be of the type {type}")
-
-        dataset_vars = {}
-
-        for each_key in ["train", "test", "validation"]:
-            if each_key in kwargs:
-                dataset = kwargs[each_key]
-                # FlyteFile
-                if issubclass(self.python_interface.inputs[each_key], FlyteFile):
-                    filepath = dataset.download()
-                    if dataset.extension() == "csv":
-                        dataset_vars[each_key] = xgboost.DMatrix(
-                            filepath
-                            + "?format=csv&label_column="
-                            + str(self._config.label_column)
-                        )
-                    else:
-                        dataset_vars[each_key] = xgboost.DMatrix(filepath)
-                # FlyteSchema
-                elif issubclass(self.python_interface.inputs[each_key], FlyteSchema):
-                    df = dataset.open().all()
-                    target = df[df.columns[self._config.label_column]]
-                    df = df.drop(df.columns[self._config.label_column], axis=1)
-                    dataset_vars[each_key] = xgboost.DMatrix(df.values, target.values)
-                else:
-                    raise ValueError(f"Invalid type for {each_key} input")
-
-        model, evals_result = self.train(
-            dtrain=dataset_vars["train"],
-            dvalid=dataset_vars["validation"] if "validation" in kwargs else None,
-        )
-        predictions = self.test(booster_model=model, dtest=dataset_vars["test"])
+        params: XGBoostParameters = kwargs[self._PARAMS_ARG]
+        test = kwargs[self._TEST_ARG]
+        train = kwargs[self._TRAIN_ARG]
+        if self._validate:
+            valid = kwargs[self._VALIDATION_ARG]
+
+        if not params.model_parameters:
+            params.model_parameters = self._config.model_parameters
+        if not params.hyper_parameters:
+            params.hyper_parameters = self._config.hyper_parameters
```

Comment:
We can remove this if default values are assigned to the XGBoostParameters dataclass.